ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/PartyManagementServiceIT.scala

```diff
@@ -20,13 +21,311 @@ import scalaz.Tag
+          party = partyNonDefault,
+          sourceIdentityProviderId = idpId1,
+          targetIdentityProviderId = idpId1,
+        )
+      )
+      get2 <- ledger.getParties(
+        GetPartiesRequest(parties = Seq(partyNonDefault), identityProviderId = idpId1)
+      )
+      // cleanup
+      _ <- ledger.updatePartyIdentityProviderId(
+        UpdatePartyIdentityProviderRequest(
+          party = partyNonDefault,
+          sourceIdentityProviderId = idpId1,
+          targetIdentityProviderId = "",
+        )
+      )
+      _ <- ledger.deleteIdentityProviderConfig(DeleteIdentityProviderConfigRequest(idpId1))
+    } yield {
+      assertEquals(
+        "default idp",
+        get1.partyDetails.map(d => d.identityProviderId -> d.party -> d.isLocal),
+        Seq("" -> partyDefault -> true),
+      )
+      assertEquals(
+        "non default idp",
+        get2.partyDetails.map(d => d.identityProviderId -> d.party -> d.isLocal),
+        Seq(idpId1 -> partyNonDefault -> true),
+      )
+    }
+  })
+
+  test(
+    "PMGetPartiesUsingDifferentIdps",
+    "Test getting parties using different idps",
+    allocate(NoParties),
+  )(implicit ec => { case Participants(Participant(ledger)) =>
+    val idpId1 = ledger.nextIdentityProviderId()
+    val idpId2 = ledger.nextIdentityProviderId()
+    for {
+      _ <- ledger.createIdentityProviderConfig(identityProviderId = idpId1)
+      _ <- ledger.createIdentityProviderConfig(identityProviderId = idpId2)
+      partyDefault <- ledger.allocateParty(identityProviderId = None).map(_.unwrap)
+      partyNonDefault <- ledger.allocateParty(identityProviderId = Some(idpId1)).map(_.unwrap)
+      partyOtherNonDefault <- ledger.allocateParty(identityProviderId = Some(idpId2)).map(_.unwrap)
+      getAsDefaultIdp <- ledger.getParties(
+        GetPartiesRequest(parties = Seq(partyDefault, partyNonDefault), identityProviderId = "")
+      )
+      getAsNonDefaultIdp <- ledger.getParties(
+        GetPartiesRequest(
+          parties = Seq(partyDefault, partyNonDefault, partyOtherNonDefault),
+          identityProviderId = idpId1,
+        )
+      )
+      // cleanup
+      _ <- ledger.updatePartyIdentityProviderId(
+        UpdatePartyIdentityProviderRequest(
+          party = partyNonDefault,
+          sourceIdentityProviderId = idpId1,
+          targetIdentityProviderId = "",
+        )
+      )
+      _ <- ledger.updatePartyIdentityProviderId(
+        UpdatePartyIdentityProviderRequest(
+          party = partyOtherNonDefault,
+          sourceIdentityProviderId = idpId2,
+          targetIdentityProviderId = "",
+        )
+      )
+      _ <- ledger.deleteIdentityProviderConfig(DeleteIdentityProviderConfigRequest(idpId1))
+    } yield {
+      assertEquals(
+        "default idp",
+        getAsDefaultIdp.partyDetails.map(_.copy(localMetadata = None)).toSet,
+        Set(
+          PartyDetails(party = partyDefault, isLocal = true, identityProviderId = ""),
```

Comment:
I don't understand why there are only 2 parties here and why the idpId is empty for the second one. It seems to me that admin won't be able to reassign the idp if it cannot read the true value of it

Reply:
We get 2 parties because we asked for only 2 parties in the request. The result for the 3rd party would be analogous to the result for the 2nd party.

Good point on the admin not being able to see the idp ids of other parties. The code that clears the idp is [this](https://github.com/DACH-NY/canton/blob/main/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/platform/apiserver/services/admin/ApiPartyManagementService.scala#L498-L500) and it depends not on a user being an admin, but on whether the user is in the default idp.

Reply:
The admin can learn the idp of a party (or user) by guessing the idp and using it in a get request. If the guess was correct then the idp value will be present in the response